### PR TITLE
added all entities as per issues and design specification

### PR DESCRIPTION
**Pull request by awesominat**
Implemented Buy/Sell/TopupTransaction, CommonUser, CommonUserFactory, CompanyInformation, CompanyNews, Pricepoint, Stock, Transaction, TransactionHistory, TransactionType, User, and UserFactory entities.

As per the issue, TopupTransaction was added, implementation of certain entities shifted to fit UseCases. For example, User was switched to have a Portfolio attribute to more closely match the view portfolio use case.

### CODE REVIEW

**Review by rickypramanick — APPROVED** [review]
These are very fundamental building blocks for our project as these are the high-level classes on which the rest of dependencies would point inwards towards. With that in mind, I believe that you have implemented the entities in a suitable manner such that we are able to expand (add other methods if needed in the projects) from this, if needed. This is a good foundation to ensure Clean Architecture and SOLID principle adherence, especially as you have maintained the practice of ensuring that there is no logic within the entities.

I think that these entities are sufficient for our proposed use cases however, I do wonder if we could include `TransactionType` as an attribute within the `Transaction` class? I was just wondering why you implemented it this way.

**Comment by quiz3** [comment]
This is a good addition. The entities are necessary for the upcoming implementation of the `use_cases`. There isn't much to say, since we are just restarting the project on a new repo, but for what it's worth, I have already reviewed the entities here on the old repo (which are the same), and fixed issues on the previous versions.

**Review by quiz3 — APPROVED** [review]
These entities are solid and well-formed. Having reviewed and edited some of the on the old repo, I can say with confidence they will make an excellent addition to the new repo. 

The `toString()` methods are a very useful feature for debugging purposes.

File: src/entities/CommonUser.java [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+    private final HashMap<String, TransactionHistory> history;
+    private Double balance;
+
+    public void updatePortfolio(String ticker, Double amount) {
+        if (amount == 0) {
+            portfolio.remove(ticker);
+            return;
+        }
+        portfolio.put(ticker, amount);
+    }
```
Comment (gursi26) [inline src/entities/CommonUser.java:19]: I think that the logic should belong to the use cases. According to the clean architecture this layer should only be comprised of classes with getter and setter functions.

**Review by gursi26 — COMMENTED** [review]
I think that the logic should belong to the use cases. According to the clean architecture this layer should only be comprised of classes with getter and setter functions.

File: src/entities/CommonUser.java (hunk 1 above)
Comment (awesominat) [reply src/entities/CommonUser.java]: I see the potential mix-up. I have changed the logic of CommonUser and User accordingly. The use case interactors must also be changed

**Pull request merged**